qt/place_page_dialog_user.cpp

```diff
@@ -0,0 +1,291 @@
+      }
+      data->addWidget(label, row++, 1);
+      return label;
+    };
+
+    if (info.IsBookmark())
+    {
+      addEntry("Bookmark", "Yes");
+    }
+
+
+    // Wikipedia fragment
+    if (auto const & wikipedia = info.GetMetadata(feature::Metadata::EType::FMD_WIKIPEDIA); !wikipedia.empty()){
+      QLabel * name = new QLabel("Wikipedia");
+      name->setOpenExternalLinks(true);
+      name->setTextInteractionFlags(Qt::TextBrowserInteraction);
+      name->setText(QString::fromStdString("<a href=\"" + feature::Metadata::ToWikiURL(std::string(wikipedia)) + "\">Wikipedia</a>"));
+      data->addWidget(name, row++, 0);
+    }
+
+    // Description
+    if (auto description = info.GetWikiDescription(); !description.empty())
+    {
+      QPushButton * wikiButton = new QPushButton("Wikipedia Description");
+      wikiButton->setAutoDefault(false);
+      connect(wikiButton, &QAbstractButton::clicked, this, [this, description, title]()
+      {
+        auto textDialog = TextDialog(this, QString::fromStdString(description), QString::fromStdString("Wikipedia: " + title));
+        textDialog.exec();
+      });
+
+      data->addWidget(wikiButton, row++, 0, 1, 2, Qt::AlignLeft);
+    }
+
+    // Opening hours fragment
+    if (auto openingHours = info.GetOpeningHours(); !openingHours.empty())
+      addEntry("Opening hours", std::string(openingHours));
+
+    // Cuisine fragment
+    if (auto cuisines = info.FormatCuisines(); !cuisines.empty())
+      addEntry("Cuisine", cuisines);
+
+    // Entrance fragment
+    // TODO
+
+    // Phone fragment
+    if (auto phoneNumber = info.GetMetadata(feature::Metadata::EType::FMD_PHONE_NUMBER); !phoneNumber.empty()){
+      data->addWidget(new QLabel("Phone"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='tel:" + std::string(phoneNumber) + "'>" + std::string(phoneNumber) + "</a>"));
+      value->setOpenExternalLinks(true);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    // Operator fragment
+    if (auto operatorName = info.GetMetadata(feature::Metadata::EType::FMD_OPERATOR); !operatorName.empty())
+      addEntry("Operator", std::string(operatorName));
+
+    // Wifi fragment
+    if (info.HasWifi()){
+      addEntry("Wi-Fi", "Yes");
+    }
+
+    // Links fragment
+    if (auto website = info.GetMetadata(feature::Metadata::EType::FMD_WEBSITE); !website.empty())
+      addEntry("Website", stripSchemeFromURI(std::string(website)), true);
+
+    if (auto email = info.GetMetadata(feature::Metadata::EType::FMD_EMAIL); !email.empty()){
+      data->addWidget(new QLabel("Email"), row, 0);
+
+      QLabel * value = new QLabel(QString::fromStdString("<a href='mailto:" + std::string(email) + "'>" + std::string(email) + "</a>"));
+      value->setOpenExternalLinks(true);
+
+      data->addWidget(value, row++, 1);
+    }
+
+    if (auto facebook = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_FACEBOOK); !facebook.empty()){
```

Comment:
As we're using BSD style for bracers, there's no need to introduce K&R one everywhere.
```suggestion
    if (auto facebook = info.GetMetadata(feature::Metadata::EType::FMD_CONTACT_FACEBOOK); !facebook.empty())
    {
```

Reply:
done